go-controller/pkg/node/gateway_init.go

```diff
@@ -107,7 +106,20 @@ func (n *OvnNode) initGateway(subnets []*net.IPNet, nodeAnnotator kube.Annotator
 		}
 	}
 
-	var err error
+	defaultGatewayIntf, defaultGatewayNextHop, err := getDefaultGatewayInterfaceDetails()
```

Comment:
I think you may need to call this after initSharedGateway. OVN-k8s will move the interfaces around (if it is not already setup correctly) so calling it here before that is done could be premature

Reply:
Not sure I follow. This patch does not change the order of these events. Today we also do `getDefaultGatewayInterfaceDetails` before `initSharedGateway`.  Is the current code wrong?

Reply:
Besides, how would that work? We need to get the primary interface before initializing stuff in shared gateway mode

Reply:
sorry you are right, I read the old code wrong.